src/include/planner/operator/sip/logical_semi_masker.h

```diff
@@ -6,37 +6,50 @@
 namespace kuzu {
 namespace planner {
 
+/*
+ * There are two types of semi mask in the system.
```

Comment:
I'm confused about the SemiMaskType notion? Regardless of how the node offsets are "collected", the semi mask is a mask of nodeOffsets, right? So it looks like SemiMaskType is really SemiMaskConstructionType or SemiMaskFillingTyp, meaning a field to identify how the semi mask is constructed (or as you call it collected but constructed or filled is more clear).